proposals/lip-introduce_validators_module.md

```diff
@@ -0,0 +1,577 @@
+
+In this LIP we specify the properties, serialization, and initialization of the validators module, as well as the protocol logic processed during a block processing and the functions exposed to other modules and to off-chain services.
+
+
+## Rationale
+
+### Generator Key
+
+To be able to create block signatures, the secret key of the validator account needs to be accessible for a generator node. The most common approach is to store the encrypted passphrase that yields the secret key (or the encrypted secret key) on the node, where the encryption key is derived from a password-based key derivation function. This results in a small security risk: If an attacker is able to get the encrypted passphrase (or encrypted secret key) and the password for the encryption key derivation, the attacker has full control over the validator account. This may be a bigger concern for validators running a Lisk node on a remote data center.
+
+
+To mitigate this risk, we propose to add an extra key pair to a validator account that is used for creating block signatures. Then, a generator node only requires access to the secret generator key, but not to the secret that is used for signing transactions. To increase security even further, validators should be able to update the generator key pair at any time. This is done by calling the `setValidatorGeneratorKey` function, described [below](#setValidatorGeneratorKey).
+
+
+### Ensuring Uniqueness of validator BLS Keys
+
+The validators module store maintains an account for each validator registered in the chain. In particular, it stores the [BLS key](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0038.md) associated with the validator, used to sign commits. BLS keys have to be unique across the chain, i.e. two validators are not allowed to register the same BLS key. To easily check whether a BLS key has been previously registered, we use a _registered BLS keys substore_, to store all the registered validator BLS keys. Store keys are set to the BLS key and the corresponding store value to the address of the validator that registered the key. This allows to check for the existence of a certain BLS key in constant time.
+
+### Block Slots
+
+In the Lisk protocol, time is divided into intervals of fixed length. 
+These intervals are called a _block slots_ and at most one block can be added to the blockchain during each slot. 
+The length of a block slot is a constant called _block time_.
+The validator module provides the functions to convert timestamp to block slots and viceversa.
```

Comment:
suggestion:
`The validator module provides the functions to calculate the block slot given the timestamp and viceversa.`

Reply:
edited